plugins/inputs/mysql/mysql.go

```diff
@@ -908,6 +932,77 @@ func (m *Mysql) gatherGlobalStatuses(db *sql.DB, serv string, acc telegraf.Accum
 		}
 	}
 
+	// gather connection metrics from user_statistics for each user
+	if m.GatherUserStatistics {
```

Comment:
aren't you doing the same thing twice here? above you check the same boolean and call it out to a separate function?